extensions/wikia/ImageLazyLoad/js/ImgLzy.module.js

```diff
@@ -67,8 +66,15 @@ define('wikia.ImgLzy', ['jquery', 'wikia.log', 'wikia.window'], function ($, log
 
 		// rewrite the URL to request WebP thumbnails (if enabled on this wiki and supported by the browser)
 		rewriteURLForWebP: function (src) {
-			if (w.wgEnableWebPThumbnails === true && this.browserSupportsWebP && thumbCheckRegExp.test(src)) {
-				src = src.replace(/\.[^\./]+$/, '.webp');
+			if (w.wgEnableWebPThumbnails === true && this.browserSupportsWebP && thumbExtCheckRegExp.test(src) && thumbnailer.isThumbUrl(src)) {
+				if (thumbnailer.isLegacyThumbnailerUrl(src)) {
+					src = src.replace(/\.[^\./]+$/, '.webp');
+				}
+				else {
+					// add "format=webp" to Vignette URL
+					src += src.indexOf('?') ? '&' : '?';
+					src += 'format=webp';
```

Comment:
I don't know of all the cases, but for example there are some places where the image is forced into a specific type from the mediawiki side (like the [AvatarService](https://github.com/Wikia/app/blob/dev/includes/wikia/services/AvatarService.class.php#L296)). 

I guess what to do depends on the use-case. When the format is changed to `jpg` because we want performance benefits, I guess webp would give us better performance anyway, right?